### ISSUE TITLE
Add global attribute for data processing level

### ISSUE THREAD

**Issue by emiliom** [OSOceanAcoustics/echopype#980]
Add a single global attribute that specifies the data processing/product level for the dataset. 

The [ACDD](http://wiki.esipfed.org/index.php/Attribute_Convention_for_Data_Discovery) convention defines the *Recommended* global attribute `processing_level`, "A textual description of the processing (or quality control) level of the data". Usage of this attribute is also recommended (possibly as part of its ACDD implementation?) in the NASA Earth Science Data and Information System Standards Office document, "Data Product Development Guide (DPDG) for Data Producers version1.1", https://doi.org/10.5067/DOC/ESO/RFC-041VERSION1

Another relevant ACDD global attribute is `product_version`, "Version identifier of the data file or product as assigned by the data creator. For example, a new algorithm or methodology could result in a new product_version." ACDD classifies it as *Suggested*.

Related to #817

**Comment by leewujung** [comment]
A minimal example of decorator to add attribute:
```python
In [29]: def test_add_attrs(func):
    ...:     def wrapper():
    ...:         da = func()
    ...:         da = da.assign_attrs({"test_attr": "some attribute"})
    ...:         return da
    ...:     return wrapper
    ...: 

In [30]: @test_add_attrs
    ...: def build_da():
    ...:     return xr.DataArray([1, 2, 3], coords={"channel": ["chA","chB","chC"]})
    ...: 

In [31]: da = build_da()

In [32]: da
Out[32]: 
<xarray.DataArray (channel: 3)>
array([1, 2, 3])
Coordinates:
  * channel  (channel) <U3 'chA' 'chB' 'chC'
Attributes:
    test_attr:  some attribute
```

**Comment by leewujung** [comment]
Out of curiosity, I tried another one with parameters:
```python
In [50]: def test_add_attrs(some_str):
    ...:     def inner(func):
    ...:        def wrapper(*args, **kwargs):
    ...:           da = func()
    ...:           da = da.assign_attrs({"input_str": some_str})
    ...:           da = da.assign_attrs({"test_attr": "some attribute"})
    ...:           return da
    ...:        return wrapper
    ...:     return inner
    ...: 

In [51]: @test_add_attrs(some_str="add this string!")
    ...: def build_da():
    ...:     return xr.DataArray([1, 2, 3], coords={"channel": ["chA","chB","chC"]})
    ...: 

In [52]: da = build_da()

In [53]: da
Out[53]: 
<xarray.DataArray (channel: 3)>
array([1, 2, 3])
Coordinates:
  * channel  (channel) <U3 'chA' 'chB' 'chC'
Attributes:
    input_str:  add this string!
    test_attr:  some attribute

```

**Comment by emiliom** [comment]
Well, thanks for doing my work for me! Adding a parameter was my first question/to-do :laughing: . Though, that inner/wrapper hierarchy plus the use of `*args, **kwargs` is kind of ugly, but that'll be a hidden ugliness.

It does look like a decorator will work. And I completely agree that from a code usability perspective, it's perfect. Hopefully it *will* work.

**Comment by leewujung** [comment]
This is the result of my pattern matching 😁 so there are probably redundancy that can be cut down, but it's fun to make it work, haha.

> Hopefully it will work.

🤞

**Comment by emiliom** [comment]
ACDD defines a couple of global attributes that end in the suffix `_url`, to specify a url corresponding to the global attribute it complements: [publisher_url](https://wiki.esipfed.org/Attribute_Convention_for_Data_Discovery_1-3#publisher_url) and [creator_url](https://wiki.esipfed.org/Attribute_Convention_for_Data_Discovery_1-3#creator_url). While it doesn't define a `processing_level_url` attribute, I thin we should add one that'll point to the compact data processing levels page we discussed.

**Comment by emiliom** [comment]
Product level attributes will be added only if lat-lon coordinates (and depth for L2 and higher?) are present.

**Comment by emiliom** [comment]
Addressed in #1001
